dataland-frontend/src/components/general/UserAuthenticationButtons.vue

```diff
@@ -0,0 +1,40 @@
+<template>
+  <div class="col-12" >
+    <Button v-if="!authenticated" label="Join" class="d-letters d-button uppercase p-button p-button-sm justify-content-center w-5rem" name="join_dataland_button" @click="register" />
+    <Button v-if="!authenticated" label="Login" class="uppercase p-button p-button-sm d-letters text-primary d-button justify-content-center bg-white-alpha-10 w-5rem ml-4" name="login_dataland_button" @click="login" />
+    <Button v-if="authenticated" label="Logout" class="uppercase p-button p-button-sm d-letters text-primary d-button justify-content-center bg-white-alpha-10 w-5rem ml-4" name="logout_dataland_button" @click="logout" />
+  </div>
+</template>
+<script>
+
+import Button from "primevue/button";
+
+export default {
+  name: 'UserAuthenticationButtons',
+  components: {Button},
+  inject: ['authenticated', 'getKeycloakInitPromise'],
```

Comment:
Kann mir jmd diese Injection des Keycloak Prmises erklären?
Habe ich oft im Code gesehe nudn würde ich gerne verstehen.
Ist das quasi das "Durchreichen" des Login-Status durch die Komponenten?

Reply:
inject macht ein provide aus der main app direkt in der Komponente verfügbar (siehe auch https://vuejs.org/guide/components/provide-inject.html#prop-drilling)